tools/mason/MasonExternal.chpl

```diff
@@ -83,22 +103,90 @@ private proc spackInstalled() throws {
   if !isDir(SPACK_ROOT) {
     throw new owned MasonError("To use `mason external` call `mason external --setup`");
   }
+  if !isDir(getSpackRegistry) {
+    throw new owned MasonError("Mason has been updated, to use `mason external` "+
+                                "call `mason external --setup`");
+  }
+  if getSpackVersion != version && SPACK_ROOT == default then {
+    throw new owned MasonError("Mason has been updated and requires a newer" +
+          " version of spack.\nTo use `mason external`, call `mason external --setup`");
+  }
   return true;
 }
 
 /* Spack installed to MASON_HOME/spack */
 proc setupSpack() throws {
   writeln("Installing Spack backend ...");
-  const destination = MASON_HOME + "/spack/";
-  const clone = "git clone -q https://github.com/spack/spack " + destination;
-  const checkout = "git checkout -q " + spackVersion;
-  const status = runWithStatus(clone);
-  gitC(destination, checkout);
-  if status != 0 {
+  // destination directory name for spack command line interface and registry
+  const destCLI = MASON_HOME + "/spack/";
+  // git clone to download spack for CLI and registry
+  const depth = ' --depth 1 ';
+  const getRepoTag = ' --branch ' + branch + ' ';
+  const repo = "https://github.com/spack/spack ";
+  const cloneCLI = "git clone -q" + getRepoTag + depth + repo + destCLI;
+  const statusCLI = runWithStatus(cloneCLI);
+  const statusPackages = cloneSpackRegistry();
+  generateYAML();
+  // check for status
+  if statusCLI != 0 && statusPackages != 0 {
     throw new owned MasonError("Spack installation failed");
+  } else {
+    writeln("Spack installation successful");
+  }
+}
+
+private proc updateSpackCommandLine() {
+  const checkOutCommand = 'git ' + '-C ' + MASON_HOME + '/spack ' +
+                      ' checkout -q FETCH_HEAD -b ' + version + '-branch';
+  const command = 'git ' + '-C ' + MASON_HOME + '/spack ' + ' fetch -q ' +
+                  ' --depth 1 origin ' + branch;
+  const commitTagCommand = 'git ' + '-C ' + MASON_HOME + '/spack ' +
+                      'tag v0.14.2 -m "updated version"';
+  const status = runWithStatus(command);
+  const statusCheckOut = runWithStatus(checkOutCommand);
+  const statusCommitTag = runWithStatus(commitTagCommand);
+  if status != 0 || statusCheckOut != 0 || statusCommitTag != 0 then return -1;
+  else return 0;
+}
+
+proc cloneSpackRegistry() {
```

Comment:
I'd advocate for adding a `/* doc-string */` to every function, even if very simple due to the procedure name being "self-documenting".